File: duplicates-android/app/src/main/java/com/github/duplicates/contact/PhoneData.java
/*
 * Source file of the Remove Duplicates project.
 * Copyright (c) 2016. All Rights Reserved.
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 2.0 (the "License"); you may not use this file except in compliance with the
 * License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/2.0
 *
 * Contributors can be contacted by electronic mail via the project Web pages:
 *
 * https://github.com/pnemonic78/RemoveDuplicates
 *
 * Contributor(s):
 *   Moshe Waisberg
 *
 */
package com.github.duplicates.contact;

import android.text.TextUtils;

import static android.provider.ContactsContract.CommonDataKinds.Phone;

/**
 * Contact telephone data.
 *
 * @author moshe.w
 */
public class PhoneData extends ContactData {

    public PhoneData() {
        setMimeType(Phone.CONTENT_ITEM_TYPE);
    }

    public String getNumber() {
        return getData1();
    }

    public int getType() {
        return Integer.parseInt(getData2());
    }

    public String getLabel() {
        return getData3();
    }

    @Override
    public String toString() {
        return getNumber() + (TextUtils.isEmpty(getLabel()) ? "" : " " + getLabel());
    }
}

File: duplicates-android/app/src/main/java/com/github/duplicates/contact/PhoneData.kt
/*
 * Copyright 2016, Moshe Waisberg
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.github.duplicates.contact

import android.text.TextUtils

import android.provider.ContactsContract.CommonDataKinds.Phone

/**
 * Contact telephone data.
 *
 * @author moshe.w
 */
class PhoneData : ContactData() {

    val number: String?
        get() = data1

    val type: Int
        get() = parseInt(data2)

    val label: String?
        get() = data3

    init {
        mimeType = Phone.CONTENT_ITEM_TYPE
    }

    override fun toString(): String {
        val label = label
        return number!! + if (TextUtils.isEmpty(label)) "" else " " + label!!
    }

    override fun containsAny(s: CharSequence): Boolean {
        return contains(label) || contains(number)
    }
}
